Mathlib/Data/Matrix/Vec.lean

```diff
@@ -76,17 +76,20 @@ theorem vec_sum [AddCommMonoid R] (s : Finset ι) (A : ι → Matrix m n R) :
   ext
   simp_rw [vec, Finset.sum_apply, vec, Matrix.sum_apply]
 
-theorem vec_dotProduct_vec [NonUnitalNonAssocSemiring R] [Fintype m] [Fintype n]
+theorem vec_dotProduct_vec [AddCommMonoid R] [Mul R] [Fintype m] [Fintype n]
```

Comment:
Please mention this generalization in the commit message, and whether it's useful or just wondering you did because you could.